scripts/setup-k8s-master.sh.in

```diff
@@ -76,20 +65,51 @@ bootstrapTokens:
   usages:
   - signing
   - authentication
-controllerManagerExtraArgs:
-  cloud-provider: "aws"
-  allocate-node-cidrs: "false"
+nodeRegistration:
+  criSocket: /var/run/dockershim.sock
+  name: "${HOSTNAME}"
+  kubeletExtraArgs:
+    cloud-provider: "aws"
+  taints:
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+---
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: ClusterConfiguration
+apiServer:
+  timeoutForControlPlane: 4m0s
+  certSANs:
+  - "${LB_IPV4}"
+  extraArgs:
+    cloud-provider: "aws"
+certificatesDir: /etc/kubernetes/pki
+clusterName: kubernetes
+controlPlaneEndpoint: "${LB_DNS}:443"
+controllerManager:
+  extraArgs:
+    cloud-provider: "aws"
+    allocate-node-cidrs: "false"
+etcd:
+  local:
+    dataDir: /var/lib/etcd
+imageRepository: k8s.gcr.io
+kubernetesVersion: "$(cat /etc/kubernetes_community_ami_version)"
+scheduler: {}
 EOF
 
 # Feature gates
-echo "featureGates:" >> /tmp/kubeadm.yaml
+echo "dns:" >> /tmp/kubeadm.yaml
 if [[ "{{ClusterDNSProvider}}" == "CoreDNS" ]]; then
-  echo "  CoreDNS: True" >> /tmp/kubeadm.yaml
+  echo "  type: CoreDNS" >> /tmp/kubeadm.yaml
+else
+  echo "  type: kube-dns" >> /tmp/kubeadm.yaml
 fi
 
 if [[ "{{NetworkingProvider}}" == "calico" ]]; then
 cat >>/tmp/kubeadm.yaml <<EOF
 networking:
+  dnsDomain: cluster.local
+  serviceSubnet: 10.96.0.0/12
```

Comment:
dnsDomain and serviceSubnet here should be the defaults and can likely be omitted.